litmus-portal/backend/graphql-server/pkg/cluster/jwt.go

```diff
@@ -0,0 +1,60 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/util"
+	"os"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+var secret = ""
+
+func ClusterCreateJWT(id string) (string, error) {
+	initSecret()
+	expirationTime := time.Now().Add(12 * time.Hour)
+	claims := jwt.MapClaims{}
+	claims["cluster_id"] = id
+	claims["exp"] = expirationTime.Unix()
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+
+	tokenString, err := token.SignedString([]byte(secret))
+	if err != nil {
+		return "", err
+	}
+	return tokenString, nil
+}
+
+func ClusterValidateJWT(token string) (string, error) {
+	initSecret()
```

Comment:
Do we need initSecret() in clusterValidateJWT. Because it was initialized by the ClusterCreateJWT function and store it in a global variable `secret`

Reply:
Yes we need it, currently we don't have a way to generate random secret keys to pass as env as the manifest for litmus-portal is static.

Reply:
But before validating the JWT of cluster it must have been generated from the above function... Thus initSecret() had already been executed before.. I don't think there is a need of executing it again

Reply:
If the server goes down and comes back up, secret will be empty and any function can be called at that time either Generate or Validate, thus we need it at both places. The init function is capable of detecting whether the secret is already set, so if it is set it will not execute.

Reply:
So instead of generating it everytime during the clusterConnect and validate. We should setup secret at the time of initializing the  server

Reply:
It's set to env var now.

Reply:
And what if env var is not set??? Are we having any default secret for that

Reply:
The server panics, it is compulsory